integration/tests/user_scenario/basic_workflow/single_env_module_last_release_deleted.rb

```diff
@@ -0,0 +1,71 @@
+require 'git_utils'
+require 'r10k_utils'
+require 'master_manipulator'
+test_name 'CODEMGMT-127 - C64120 - Attemp to Deploy Environment with Forge Module where Latest Release has been Deleted'
+
+#This test uses the spotty module at https://forge-aio01-petest.puppetlabs.com//puppetlabs/spotty, which has valid 0.1.0 and 0.3.0 versions, and deleted 0.2.0 and 0.4.0 versions.
+
+#Init
+env_path = on(master, puppet('config print environmentpath')).stdout.rstrip
+prod_env_modules_path = File.join(env_path, 'production', 'modules')
+
+git_environments_path = '/root/environments'
+last_commit = git_last_commit(master, git_environments_path)
+
+#Verification
+module_contents = 'Version 3'
+module_contents_regex = /\A#{module_contents}\n\z/
+
+module_contents_path = File.join(prod_env_modules_path, 'spotty', 'manifests', 'init.pp')
+module_version_filepath = File.join(prod_env_modules_path, 'spotty', 'metadata.json')
+module_version_3_regex = /"0.3.0"/
+
+#File
+puppet_file = <<-PUPPETFILE
+mod "puppetlabs/spotty"
+PUPPETFILE
+
+puppet_file_path = File.join(git_environments_path, 'Puppetfile')
+
+#Teardown
+teardown do
+  clean_up_r10k(master, last_commit, git_environments_path)
+
+  step 'Remove module'
+  on(agents, "rm -rf #{module_contents_path}")
```

Comment:
This is technically not required because "clean_up_r10k" already will erase the target directory.